webapp/src/main/java/org/apache/atlas/notification/NotificationHookConsumer.java

```diff
@@ -443,6 +453,27 @@ public int getHandlerOrder() {
         return HandlerOrder.NOTIFICATION_HOOK_CONSUMER.getOrder();
     }
 
+    public void closeImportConsumer(String importId, String topic) {
+        try {
+            LOG.info("==> closeImportConsumer(importId={}, topic={})", importId, topic);
+            String consumerName = ATLAS_IMPORT_CONSUMER_THREAD_PREFIX + importId;
+            ListIterator<HookConsumer> consumersIterator = consumers.listIterator();
+            while (consumersIterator.hasNext()) {
+                HookConsumer consumer = consumersIterator.next();
+                if (consumer.getName().contains(consumerName)) {
+                    consumer.shutdown();
+                    consumersIterator.remove();
+                }
+            }
+            notificationInterface.closeConsumer(ASYNC_IMPORT);
```

Comment:
This would close all consumers of type `ASYNC_IMPORT`. Shouldn't only consumers of `topic` be closed instead? Perhaps only one consumer will be active for type `ASYNC_IMPORT` in the current implementation status; but this part of code should better handle multiple simultaneous imports.